Cargo.toml

```
@@ -0,0 +1,10 @@
+[package]
+name = "mixxx"
+version = "2.3.0"
+authors = ["Mixxx DJ Software maintainers"]
+edition = "2018"
+
+[build-dependencies]
+# sccache's handling of the /fp:fast MSVC compiler option is broken, but this is fixed in the master branch.
+# https://github.com/mozilla/sccache/pull/962
+sccache = { git = "https://github.com/mozilla/sccache.git", branch = "master" }
```

```suggestion
sccache = { git = "https://github.com/mozilla/sccache.git", branch = "master" }

[lib]
path = "non-existent" # required so `cargo update` runs

```
If you add this, you can remove the `main.rs` file. works on my machine but try yourself as well please.

---

cool, thanks for this hack

---

This lets `cargo update` work but I think what we actually need on CI is `cargo build`.

---

Can't say much about that. I don't understand what this PR is supposed to do anyways. I just figured there must be a better way than leaving a dummy `main.rs` in the repo.

---

The purpose of this PR is to use Cargo.lock as the cache key instead of the ugly hack of a git commit hash which won't work well as we add more Rust dependencies.

---

Right, but why do we need to change `cargo install --git https://github.com/mozilla/sccache.git --branch master`

---

Sorry for asking stupid questions. The solution seems overcomplicated to me, but thats usually a sign that I don't know enough about the actual topic.

---

The current solution works for now. But it won't work when we also have other Rust dependencies to cache.